Add SimpleCov to measure code coverage of Action Mailer

aaronang opened the pull request:
### Summary

Currently, Ruby on Rails does not measure code coverage, which can be a useful tool to see what is or is not tested. Therefore, in this patch, we propose the addition of SimpleCov. With SimpleCov, we can easily get an overview of the coverage for each file. Furthermore, we have tried to make it as elegant as possible by defining a Rake task, such that we can get the coverage data by running the following command from the Action Mailer directory: `bundle exec rake coverage`. The code coverage analysis results will be placed in a `coverage` directory, which will be created in the root of the Action Mailer module.
### Other information

We, @ValMai, @vliegenthart, @basvijzendoorn, and myself, think it would be a nice addition to Rails to use a code coverage analysis tool. We would like to discuss with you what you think of this proposal.

Furthermore, we have made some changes to show a possible integration of a code coverage analysis tool, namely [SimpleCov](https://github.com/colszowka/simplecov). Ultimately, we would like to implement SimpleCov, such that code coverage analysis results are generated during the build and test process by Travis CI.

We are looking forward to hearing from you :smile:

aaronang committed:
Add SimpleCov to measure code coverage of Action Mailer

Code coverage is a useful tool to see what is or is not tested.
Therefore, in this patch, SimpleCov is added to measure code
coverage. The coverage data can be retrieved by running: `bundle
exec rake coverage`.

vm00z commented:
Currently, Ruby on Rails does not measure **code** coverage, which can be a useful tool to see what is **or** is not tested. Therefore, in this patch, we propose the addition of SimpleCov. With SimpleCov, we can easily get an overview of the coverage for each file. Furthermore, we have tried to make it as elegant as possible by defining a Rake task, such that we can get the coverage data by running the following command in root: `bundle exec rake coverage`.

Something like that?

tjschuck committed:
Version 3.1.11 works on Windows again

This undoes https://github.com/rails/rails/commit/7241498e51120b9847a8bc16cf48551db0f3e216

https://github.com/codahale/bcrypt-ruby/issues/128 is fixed and closed.

rafaelfranca committed:
Merge pull request #24144 from tjschuck/bcrypt_on_windows_fixed

Version 3.1.11 works on Windows again

rafaelfranca committed:
Fix Gemfile.lock

vliegenthart commented:
If the commit message is a bit more descriptive it's all good :)

kaspth committed:
Remove unfinished command infrastructure.

If we're gonna do this right, it will look mighty different from this anyway.
(Looking at you, Rails 5.1).

It isn't being used in any code as of now, so yanking is the best option.

ghost commented:
Wow

ghost commented:
We, @ValMai, @vliegenthart, @basvijzendoorn, and myself, are wondering why Ruby on Rails does not use any code coverage analysis tool. We would like to discuss with you the reasoning behind this.

Change into

We, @ValMai, @vliegenthart, @basvijzendoorn, think using a code coverage analysis tool would be a great benefit for Rails. We would like to discuss with you the reasoning behind this.

vm00z commented:
I think we should add an explanation where they can find the coverage folder after the tool. Or just explain what the command exactly does. 
Also I think the Other Information part is good as is and should not be rephrased.

ghost commented:
I don't like the word "wondering". As if they had it wrong or something.

vm00z commented:
Change 
... following command in root:
to 
... following command from the Action Mailer directory:

vm00z commented:
Maybe change to:
We, @ValMai, @vliegenthart, @basvijzendoorn, and myself, think it would be a nice addition to Rails to use a code coverage analysis tool. We would like to discuss with you what you think of this proposal.
